src/backend/distributed/executor/multi_utility.c

```diff
@@ -3727,7 +3736,13 @@ PlanGrantStmt(GrantStmt *grantStmt)
 			}
 			isFirst = false;
 
-			Assert(priv->cols == NIL);
+			if (priv->cols != NIL)
+			{
+				ereport(ERROR, (errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
```

Comment:
have we established that we're dealing with distributed tables at this point?

Reply:
correct, only distributed tables are returned from ```CollectGrantTableList```